go/vt/vtgate/planner.go

```diff
@@ -102,6 +102,7 @@ func (plr *Planner) WatchVSchema(ctx context.Context) {
 		plr.mu.Lock()
 		plr.vschema = vschema
 		plr.mu.Unlock()
+		plr.plans.Clear()
```

Comment:
I agree, it's more complicated. With this, there can be plans built with the new vschema, that get flushed for nothing, but that's it. I didn't want to double-lock, then we have to guarantee ordering and such.